common/schema-builder/src/main/java/io/smallrye/graphql/schema/helper/RolesAllowedDirectivesHelper.java

```diff
@@ -0,0 +1,78 @@
+package io.smallrye.graphql.schema.helper;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+
+import io.smallrye.graphql.schema.Annotations;
+import io.smallrye.graphql.schema.model.DirectiveArgument;
+import io.smallrye.graphql.schema.model.DirectiveInstance;
+import io.smallrye.graphql.schema.model.DirectiveType;
+import io.smallrye.graphql.schema.model.Reference;
+import io.smallrye.graphql.schema.model.Scalars;
+
+public class RolesAllowedDirectivesHelper {
+
+    public final static DirectiveType ROLES_ALLOWED_DIRECTIVE_TYPE;
+
+    static {
+        ROLES_ALLOWED_DIRECTIVE_TYPE = new DirectiveType();
+        ROLES_ALLOWED_DIRECTIVE_TYPE.setName("rolesAllowed");
+        Set<String> locations = new LinkedHashSet<>();
+        locations.add("FIELD_DEFINITION");
+        ROLES_ALLOWED_DIRECTIVE_TYPE.setLocations(locations);
+        ROLES_ALLOWED_DIRECTIVE_TYPE.setDescription("Used to specify the role required to execute a given field or operation.");
+        ROLES_ALLOWED_DIRECTIVE_TYPE.setRepeatable(false);
+
+        addArgument("value", Scalars.getStringScalar());
+    }
+
+    private static void addArgument(String name, Reference reference) {
+        DirectiveArgument arg = new DirectiveArgument();
+        arg.setName(name);
+        arg.setReference(reference);
+        ROLES_ALLOWED_DIRECTIVE_TYPE.addArgumentType(arg);
+    }
```

Comment:
```suggestion

        ROLES_ALLOWED_DIRECTIVE_TYPE.addArgumentType(createArgument("value", Scalars.getStringScalar());
    }

    private static void createArgument(String name, Reference reference) {
        DirectiveArgument arg = new DirectiveArgument();
        arg.setName(name);
        arg.setReference(reference);
        return arg;
    }
```

Makes easier to follow the flow, because all writing to the constant happens in a single method.